#!/usr/bin/python

PACKAGE="cascaded_pid_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
# physical parameters
gen.add("mass", double_t, 0, "Mass of the drone.", 1.0, 0.1, 3.0)
gen.add("min_thrust", double_t, 0, "Lower bound of collective thrust the motors at least generate.", 4.0, 0, 100)
gen.add("max_thrust", double_t, 0, "Upper bound of collective thrust the motors can generate.", 50.0, 10, 100)

gen.add("xy_same_params", bool_t, 0, "Use same parameters for lateral controlling. (X & Y control)", True)
# params for X
gen.add("kp_x", double_t, 0, "P gain of lateral position control in north (X control).", 3.0, 0.0, 100.0)
gen.add("kd_x", double_t, 0, "D gain of lateral position control in north (X control).", 2.0, 0.0, 100.0)
gen.add("max_abs_accel_x", double_t, 0, "Maximum commandable acceleration in north (absolute value in X direction). The output command will be within [-max_abs_accel_x, max_abs_accel_x].", 10.0, 0.1, 50.0)

# params for Y
gen.add("kp_y", double_t, 0, "P gain of lateral position control in west (Y control).", 3.0, 0.0, 100.0)
gen.add("kd_y", double_t, 0, "D gain of lateral position control in west (Y control). ", 2.0, 0.0, 100.0)
gen.add("max_abs_accel_y", double_t, 0, "Maximum commandable acceleration in west (absolute value in Y direction). The output command will be within [-max_abs_accel_y, max_abs_accel_y].", 10.0, 0.1, 50.0)

# params for Z
gen.add("kp_z", double_t, 0, "P gain of altitude control (Z control).", 5.5, 0.0, 100.0)
gen.add("kd_z", double_t, 0, "D gain of altitude control (Z control).", 2.2, 0.0, 100.0)

gen.add("rp_same_params", bool_t, 0, "Use same parameters for roll and pitch controlling.", True)
# params for roll
gen.add("kp_roll", double_t, 0, "P gain of roll of attitude control.", 3.0, 0.0, 100)

# params for pitch
gen.add("kp_pitch", double_t, 0, "P gain of pitch of attitude control.", 3.0, 0.0, 100)

# params for yaw
gen.add("kp_yaw", double_t, 0, "P gain of yaw of yaw control.", 1.0, 0.0, 100)

exit(gen.generate(PACKAGE, "CascadedPid", "CascadedPid"))
